src/palace/manager/sqlalchemy/model/__init__.py

```diff
@@ -33,3 +39,29 @@
 import palace.manager.sqlalchemy.model.saml
 import palace.manager.sqlalchemy.model.time_tracking
 import palace.manager.sqlalchemy.model.work
+from palace.manager.sqlalchemy.model.base import Base
+from palace.manager.sqlalchemy.model.collection import Collection
+from palace.manager.sqlalchemy.model.integration import (
+    IntegrationConfiguration,
+    IntegrationLibraryConfiguration,
+)
+from palace.manager.sqlalchemy.model.library import Library
+
+# The following supports an optimization in `Library.active_collections`.
+
+
+@event.listens_for(IntegrationConfiguration, "after_insert")
+@event.listens_for(IntegrationConfiguration, "after_delete")
+@event.listens_for(IntegrationConfiguration, "after_update")
+@event.listens_for(IntegrationLibraryConfiguration, "after_insert")
+@event.listens_for(IntegrationLibraryConfiguration, "after_delete")
+@event.listens_for(IntegrationLibraryConfiguration, "after_update")
+def handle_collection_change(_: Mapper, _connection: Connection, target: Base) -> None:
+    Library.clear_active_collections_cache(target)
```

Comment:
Minor: Is it possible to move these to `listeners.py` with the rest of the event listeners?

Reply:
Good suggestion. Thanks! Done.